pylint/reporters/progress_reporters.py

```diff
@@ -0,0 +1,74 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/pylint-dev/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/pylint-dev/pylint/blob/main/CONTRIBUTORS.txt
+
+import abc
+from typing import Optional
+
+
+class BaseProgressReporter:
+    """Progress reporter."""
+
+    def __init__(self) -> None:
+        self.item_count: Optional[int] = None
+        self.current_count = 0
+
+    def start_get_asts(self) -> None:
+        """Show message, starting to get all ASTs."""
+
+    def start_linting(self) -> None:
+        """Show message, starting linting."""
```

Comment:
Should these also be abstract?

Reply:
I personally would not as these just define the interface and subclasses can choose to override. The Null reporter doesn’t need to do not making it abstract means less code overall.

Reply:
Updated class def per later feedback.